### PR TITLE
Introduce new OpenTelemetryTelemetryClient and AggregateTelemetryClient

### PR DESCRIPTION
**Pull request by lgostling**
# Changes

### OpenTelemetryTelemetryClient
implements ITelemetryClient and emits OpenTelemetry events as Spans.

### AggregateTelemetryClient
implements ITelemetryClient, constructed of multiple ITelemetryClient objects and emits telemetry to each. This will be used to Aggregate ApplicationInsight and OpenTelemetry telemetry clients during migration to Genva.

### HttpClientWrapper
remove trace from http requests as these will be tracked through telemetry.

### ApplicationInsightsTelemetryClient
remove logging from telemetry client, this behavior is redundant with modern telemetry.

### OpenTelemetryTracer
add `GetActivity(string name)` to allow OpenTelemetryTelemetryClient to create custom named Spans.

### CODE REVIEW

**Review by varshavadaga — APPROVED** [review]

File: Core.Telemetry/ApplicationInsightsTelemetryClient.cs [hunk 1]
```diff
@@ -79,7 +80,7 @@ public void LogWarning(string message, IDictionary<string, string> additionalPro
             {
                 foreach (KeyValuePair<string, string> property in additionalProperties)
                 {
-                    if ( properties.ContainsKey(property.Key) )
+                    if (properties.ContainsKey(property.Key))
```
Comment (AllainPL) [inline Core.Telemetry/ApplicationInsightsTelemetryClient.cs:83]: I would replace the whole if else block with
```csharp
properties[property.Key] = property.Value;
```

Comment (AllainPL) [reply Core.Telemetry/ApplicationInsightsTelemetryClient.cs:83]: and while at it do the same in line 104

**Review by AllainPL — APPROVED** [review]
Apart of the minor comments looks fine 👍

File: Core.Telemetry/OpenTelemetryTelemetryClient.cs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            }
+
+            string identityToTrack = identity;
+            bool guidIdentity = Guid.TryParse(identityToTrack, out Guid _);
+            if (guidIdentity)
+            {
+                identityToTrack = identityToTrack.Substring(0, 4); // For security reasons, if the identity is a GUID, only capture the first 4 characters in the telemetry.
+            }
+
+            trace.AddTag("ApiName", apiName);
```
[the remark this answers is not in the document]
Comment (lgostling) [reply Core.Telemetry/OpenTelemetryTelemetryClient.cs:151]: This is no big deal to add additional tags later, they do not cause any problems.

**Pull request merged**